pkg/upgrade/upgrade.go

```diff
@@ -56,200 +63,203 @@ const CredentialsSecretName string = "weave-gitops-enterprise-credentials"
-	tempDir, err := ioutil.TempDir("", "git-")
-	if err != nil {
-		return fmt.Errorf("failed to create temp folder for remote git clone of profiles repo: %w", err)
 	}
 
-	defer os.RemoveAll(tempDir)
+	return []runtime.Object{helmRepository, helmRelease}
+}
 
-	_, err = gitClient.Clone(
-		ctx,
-		tempDir,
-		enterpriseProfileURL,
-		uc.Profile.ProfileConfig.ProfileBranch,
-	)
-	if err != nil {
-		return fmt.Errorf("failed to clone git repo: %v", err)
-	}
+// func makeKubeClient() (client.Client, error) {
+// 	scheme := runtime.NewScheme()
+// 	schemeBuilder := runtime.SchemeBuilder{
+// 		sourcev1.AddToScheme,
+// 		corev1.AddToScheme,
+// 	}
 
-	profileDefinition := uc.Profile
-	profileDefinition.URL = "file://" + tempDir
+// 	err := schemeBuilder.AddToScheme(scheme)
+// 	if err != nil {
+// 		return nil, fmt.Errorf("error adding sourcev1 to kube client scheme %v", err)
+// 	}
 
-	err = addProfile(uc.InstallConfig, profileDefinition, uc.CLIGitConfig)
-	if err != nil {
-		return err
-	}
+// 	kubeClientConfig := config.GetConfigOrDie()
 
-	if dryRun {
-		return nil
-	}
+// 	return client.New(kubeClientConfig, client.Options{Scheme: scheme})
+// }
 
-	err = catalog.CreatePullRequest(pctlSCMClient, pctlGitClient, uc.SCMConfig.Branch, uc.InstallConfig.RootDir)
+// buildUpgradeConfigs sets some flags default values from the env
+func buildUpgradeConfigs(ctx context.Context, uv UpgradeValues, kubeClient client.Client, gitClient git.Git, w io.Writer) (*UpgradeValues, error) {
+	repoUrlString, err := gitClient.GetRemoteUrl(".", uv.Remote)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	return nil
-}
+	uv.RepoURL = repoUrlString
 
-// addProfile installs the profile into a local git repo.
-func addProfile(installConfig install.Config, profile catalog.Profile, cliGitConfig pctl_git.CLIGitConfig) error {
-	manager := &catalog.Manager{}
+	// Calculate defaults from current working directory
+	if uv.RepoOrgAndName == "" {
+		githubRepoPath, err := getRepoOrgAndName(repoUrlString)
+		if err != nil {
+			return nil, err
+		}
 
-	// We're not interested in mocking this out for now (it just runs git commands on the local fs) so we make an inline client.
-	gitClient := pctl_git.NewCLIGit(cliGitConfig, &runner.CLIRunner{})
-	installConfig.GitClient = gitClient
+		fmt.Fprintf(w, "Deriving org/repo for PR as %v\n", githubRepoPath)
+		uv.RepoOrgAndName = githubRepoPath
+	}
 
-	return manager.Install(catalog.InstallConfig{
-		Clients: catalog.Clients{Installer: install.NewInstaller(installConfig)},
-		Profile: profile,
-	})
-}
+	if uv.GitRepository == "" {
+		gitRepositoryNameNamespace := fmt.Sprintf("%s/%s", uv.Namespace, strings.TrimSuffix(filepath.Base(repoUrlString), ".git"))
+		fmt.Fprintf(w, "Deriving name of GitRepository Resource as %v\n", gitRepositoryNameNamespace)
+		uv.GitRepository = gitRepositoryNameNamespace
+	}
```

Comment:
I think we can remove this now as it was only important for profiles.

+ also rm the `ensureGitRepositoryResource` and `getGitRepositoryNamespaceAndName` too

Reply:
👌 makes sense 🙏